### PR TITLE
[ MB-15054 ] Edit Net Weight for Service Counselors

### PR DESCRIPTION
**Pull request by rogeruiz**
## Summary

Implement the edit PPM net weights component (#10122) in the PPM review
documents page to allow for Service Counselors to edit the net weight and the
net weight remarks of a PPM shipment.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use two separate terminals to test this locally.</summary>

##### Terminal 1

Start the UI locally.

```sh
make client_run
```

##### Terminal 2

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Navigate to a Closeout PPM shipment Move as a Service Counselor in the
   Office Application.
1. Navigate to the Details page for the Move.
1. Click on _Review documents_.
1. Note the new EditPPMNetWeights component under the _Final weight_.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Test this in the ephemeral environment and please follow the **Additional
  steps** section from above.

### CODE REVIEW

**Review by AnnaGingle — DISMISSED** [review]
Need 16px spacing between remarks net weight. 

If the move does not have excess weight & no edits have been made original weight shouldn't show.

https://www.figma.com/file/5AhuEkaH3bJVrtffU3wp22/Milmove-Office-App?node-id=9897-115794&t=stct3gIaHwGlcKe7-4

File: src/components/Office/PPM/EditNetWeights/EditPPMNetWeight.jsx [hunk 1]
```diff
@@ -86,80 +90,106 @@ const WeightCalculationHint = ({ type, firstValue, secondValue, thirdValue }) =>
...
+const EditPPMNetWeightForm = ({ onCancel, initialValues, weightTicket }) => {
+  const queryClient = useQueryClient();
+
+  const { mutate: patchWeightTicketMutation } = useMutation({
+    mutationFn: patchWeightTicket,
+    onSuccess: () => {
+      queryClient.invalidateQueries({
+        queryKey: [DOCUMENTS],
+      });
+      onCancel();
```
Comment (afoon) [inline src/components/Office/PPM/EditNetWeights/EditPPMNetWeight.jsx:106]:
```suggestion
      onCancel();
```
This should happed where on the callback where `patchWeightTicketMutation` is called.

**Comment by rogeruiz** [comment]
@dinabeigelman on Monday, could you help me clarify @AnnaGingle's comment? I want to make sure I get the UI correct here, but the Figma link and the comment don't make sense to me.

Also, this is ready for review now @transcom/truss-pamplemoose. Thanks again for all your help @afoon.

**Comment by rileyorr** [comment]
Hi @rogeruiz! Here's some additional context based on Anna's feedback: 

> Need 16px spacing between remarks net weight

The padding between the updated weight value and the remarks should be 16px. I've attached screenshots of what I saw in local vs Figma for reference. 

Additionally, the padding between the original weight value and the updated weight value should be 8px. The padding was set to 16px when I checked local. 

Local:
Screenshot 2023-03-20 at 12 12 29 PM


Figma:
Screenshot 2023-03-20 at 12 09 26 PM

> If the move does not have excess weight & no edits have been made original weight shouldn't show.

I grabbed a screenshot from one of the moves as an example to talk through my understanding of this. In this example if the Service Counselor navigates to review documents and this move does not have excess weight and the weight has not been edited, the `4,000 lb | original weight` wouldn't display. It would only appear if the weight was edited. 

However, I don't see that design in [the PPM NetWeight behavior logic art board](https://www.figma.com/file/5AhuEkaH3bJVrtffU3wp22/Milmove-Office-App?node-id=8643%3A101599&t=VX4of2kKFdOOh4Us-1) so maybe I'm misunderstanding it. @dinabeigelman do you have additional insight into this? 

Example screenshot: 
Screenshot 2023-03-20 at 12 21 39 PM

Please let me know if you have any questions!

**Comment by rogeruiz** [comment]
@rileyorr can you take another look at this? Thank you.

**Comment by rileyorr** [comment]
@rogeruiz thank you for making those updates!

I did another review and noticed styling issues for when a PPM has excess weight. I added a recording below of the following things I noticed:
- There should be 24px margin between the 'Full weight' input field and the 'Edit PPM net weight' header 
- After selecting the 'edit' button, the excess weight calculation is missing the 24px padding between the 'Edit PPM net weight' header and the 'move weight - weight allowance = excess weight' calculation
- The divider line should be above 'excess weight' and only extend across the width of the calculation container ([designs for reference](https://www.figma.com/file/5AhuEkaH3bJVrtffU3wp22/Milmove-Office-App?node-id=8643%3A101599&t=H6wtMMl99KQpC1Jw-1))

https://user-images.githubusercontent.com/46329147/226630869-f1708eb2-ee4a-4514-b141-44a568c9de7f.mov

Please let me know if you have any questions!

**Comment by rogeruiz** [comment]
Thanks @rileyorr for the feedback. I believe I've addressed everything that I saw in the video and compared it against the Figma designs.

**Comment by rogeruiz** [comment]
There's a new breaking test in here that I'll address after lunch. But otherwise, this is looking good to merge after an approval from @rileyorr on this.

**Comment by khayenis** [comment]
@rogeruiz covering for Riley/Anna while they're out. The updates you made look good 👍.

One separate thing I noticed: it looks like the design calls for 8px padding between the "remarks" header and entered text. Right now, there doesn't seem to be any padding in the local environment. Let me know if this makes sense:

Design ([Figma](https://www.figma.com/file/5AhuEkaH3bJVrtffU3wp22/Milmove-Office-App?node-id=9897-115794&t=S5NWzzcm2DbsxVie-0))

Screenshot 2023-03-21 at 4 19 16 PM


Local:
Screenshot 2023-03-21 at 4 04 48 PM

cc @rileyorr

**Comment by rogeruiz** [comment]
Thanks @khayenis this should be fixed now.

File: src/components/Office/PPM/EditNetWeights/EditPPMNetWeight.jsx [hunk 2]
```diff
@@ -210,4 +252,10 @@ const EditPPMNetWeight = ({ netWeightRemarks, weightTicket, weightAllowance, shi
   );
 };
 
+EditPPMNetWeight.propTypes = {
```
Comment (mmayo-truss) [inline src/components/Office/PPM/EditNetWeights/EditPPMNetWeight.jsx:255]: Good call to add propTypes

File: src/components/Office/PPM/EditNetWeights/EditPPMNetWeight.jsx [hunk 3]
```diff
@@ -86,85 +92,120 @@ const WeightCalculationHint = ({ type, firstValue, secondValue, thirdValue }) =>
...
+    mutationFn: patchWeightTicket,
+    onSuccess: () => {
+      queryClient.invalidateQueries({
+        queryKey: [DOCUMENTS],
+      });
+    },
+  });
+
+  /**
+   * @const onSubmit
```
Comment (mmayo-truss) [inline src/components/Office/PPM/EditNetWeights/EditPPMNetWeight.jsx:112]: These JS docs comments will be really helpful for other engineers going forward.

**Comment by khayenis** [comment]
@rogeruiz  Looks good to me. Thanks!

**Review by khayenis — APPROVED** [review]
This is approved

**Pull request merged**